src/middlewares/multipartFormDataParser.js

```diff
@@ -0,0 +1,49 @@
+const BusBoy = require('busboy')
+const contentTypeLib = require('content-type')
+const createError = require('http-errors')
+
+module.exports = () => ({
+  before: async (handler, next) => {
+    const { headers } = handler.event
+    if (!headers) {
+      return next()
+    }
+
+    const contentType = headers['content-type']
+    if (contentType) {
+      const { type } = contentTypeLib.parse(contentType)
+      if (type === 'multipart/form-data') {
+        try {
+          handler.event.body = await parseMultipartData(handler.event)
+        } catch (err) {
+          throw new createError.UnprocessableEntity('Invalid or malformed multipart/form-data was provided.')
+        }
+      }
+    }
+  }
+})
+
+const parseMultipartData = async (event) => {
+  let multipartData = {}
+  const bb = BusBoy({ headers: event.headers })
+
+  return new Promise((resolve, reject) => {
+    bb
+      .on('file', (fieldname, file, filename, encoding, mimetype) => {
+        let attachment = {
+          filename,
+          mimetype,
+          encoding
+        }
+
+        file.on('data', data => { attachment.content = data })
```

Comment:
Not sure about this one.

Does `BusyBoy` emit only one chunk when all the data is converted or could emit multiple times as soon as there's some new chunk of data available (as any common stream implementation)?

In the latter case, we should somehow concatenate the new chunks to the content as soon as they are emitted rather than just assigning.

Also, would `content` be a buffer or a string?

Reply:
This was something I was also unsure of. I have this middleware implemented in my current project and I have only ever seen the data event fire once for a file. Changed this section to keep track of the chunks and their length, and on the end event use `Buffer.concat`

Reply:
I was finally able to have a look at `busboy` (which I am quite liking, by the way... 😛 ) and it seems that it is implemented as a `WritableStream`, which means that if you have a reasonably big input in a form (imagine a big picture > 16kb in most cases), you could see multiple chunks being emitted, so you have to concatenate them accordingly.

I'll try to review this better and see if the code needs to be adjusted accordingly

Reply:
I am also realizing that the way we are implementing this is "eager", meaning that we are loading all the content of the form in memory as busboy is parsing it.

This is generally not a big deal, and it's kind of an expected pattern in lambda (since we have to construct a single big object that represent all data for the current event).

Anyway this approach is generally an anti-pattern especially when managing big files, so I'll make sure to drop a note about this in the README, just to raise awareness with the users about the lambda trade-offs we are making here.

Reply:
@lmammino I know I'm chiming in on an already merged PR, but I have actually been working on middy + busboy just last month, and maybe some of the things I learned might be of interest. Currently there are some strict limits on file sizes for both lambdas ([max. 6 MB invocation payload](https://docs.aws.amazon.com/lambda/latest/dg/limits.html)) as well as API Gateway ([max. 10 MB payload size](https://docs.amazonaws.cn/en_us/apigateway/latest/developerguide/limits.html)). In the end we abandonend the approach and now use an Elastic Beanstalk server. The 6MB limit might be of interest for anyone trying to use lambda with form-data.

Reply:
Good point. This is what I ended up writing for the docs. Feel free to submit a PR if you feel it's worth a redaction or an addition :)